opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/TracingFilter.java

```diff
@@ -217,7 +217,9 @@ public void onStartAsync(AsyncEvent event) throws IOException {
                         }
                     });
                 }
-                span.deactivate();
+                if (!httpRequest.isAsyncStarted()) {
```

Comment:
Originally tried using this in the `startActive`, but it doesn't return an appropriate value at that point - so the issue is that currently this parent scope is not closed. So this needs to be reported as feedback on the OT java API RC. A possible approach may be to have an overloaded close that allows the finishSpan to be specified at the close - for cases where the decision is unknown when the scope is started.

Reply:
Just to be clear - does this currently work by checking `httpRequest.isAsyncStarted()` and then closing the span?

Reply:
We should close the scope for sync and async. The thread no longer belongs to the request. 

I suggest using `finishOnClose = false` when starting the span. Then in async callbacks use `finishOnClose = true`.

 Here is the patch https://pastebin.com/JT67yUYH

Reply:
Updated to use the approach discussed - so when async/sync not know when the Scope is started - it starts with 'finishOnClose=false' and then explicitly closes the span associated with the Scope if turns out to be synchronous.